src/flint/fq_default_mpoly.jl

```diff
@@ -567,19 +579,25 @@ function (R::FqDefaultMPolyRing)()
     return fq_default_mpoly(R, R.data())
 end
 
-function (R::FqDefaultMPolyRing)(b::Union{fq_default, Integer, fmpz})
-    return fq_default_mpoly(R, R.data(base_ring(R)(b)))
+function (R::FqDefaultMPolyRing)(b::IntegerUnion)
+    return fq_default_mpoly(R, R.data(b))
 end
 
+function (R::FqDefaultMPolyRing)(b::fq_default)
+    parent(b) == base_ring(R) || error("Unable to coerce element")
+    return fq_default_mpoly(R, R.data(_unchecked_coerce(base_ring(R.data), b)))
+end
+
+
 function (R::FqDefaultMPolyRing)(a::fq_default_mpoly)
    parent(a) == R || error("Unable to coerce polynomial")
    return a
 end
 
-function (R::FqDefaultMPolyRing)(a::Vector{fq_default}, b::Vector{Vector{T}}) where {T <: Union{fmpz, UInt}}
-    F = base_ring(R)
-    A = elem_type(F)[F(ai) for ai in a]
-    return fq_default_mpoly(R, R.data(A, b))
+function (R::FqDefaultMPolyRing)(a::Vector{fq_default}, b::Vector{Vector{Int}})
+    F = base_ring(R.data)
+    ad = elem_type(F)[_unchecked_coerce(F, ai) for ai in a]
```

Comment:
Why is there a coercion here? Shouldn't this throw if the parents don't match?

Reply:
The coersion is from fq_default to nmod/fq_nmod/fq as chosen by base_ring(R.data). The parent checks are a separate issue and can be added if you want.

Reply:
happy now?